### PR TITLE
Update/routing

### PR DESCRIPTION
**Pull request by Hamdrive**
## Overview

Successfully implemented and updated the route and auth/data handling for the entire app

Changes:
1.  Updated route handling for loader
2. Remove JSON conversion not needed as per comment in #7
3. Implementation of working test credentials for easy login
4. Fix updation of user displayname

### CODE REVIEW

**Review by shivampandey0 — APPROVED** [review]
Great to see proper error handling to show user messages for errors that occurred in fetching data.

Good to merge

File: src/context/data-context.jsx [hunk 1]
```diff
@@ -63,8 +65,13 @@ const DataProvider = ({ children }) => {
       try {
         setLoading(true);
         const newTweetorData = await fetchTwitterUser(username, dataDispatch);
-        if (newTweetorData) {
-          const userIdRef = JSON.parse(localStorage.getItem("userID"));
+        console.log(newTweetorData?.errors?.[0]?.title);
```
Comment (prachi-sahani) [inline src/context/data-context.jsx]: can be removed

File: src/context/data-context.jsx [hunk 2]
```diff
@@ -63,8 +65,13 @@ const DataProvider = ({ children }) => {
...
+        if (newTweetorData?.errors?.[0]?.title === "Not Found Error") {
+          searchError.current = `Could not find user with ${newTweetorData?.errors?.[0]?.value}`;
+          console.log(searchError.current);
```
Comment (prachi-sahani) [inline src/context/data-context.jsx]: can be removed

**Review by prachi-sahani — COMMENTED** [review]
Overall the code looks good
Good error handling
Good to merge

File: src/context/data-context.jsx (hunk 1 above)
Comment (Hamdrive) [reply src/context/data-context.jsx]: Done

File: src/context/data-context.jsx (hunk 2 above)
Comment (Hamdrive) [reply src/context/data-context.jsx]: Done

**Pull request merged**